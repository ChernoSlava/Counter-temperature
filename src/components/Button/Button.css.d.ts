export const Btn: string;
export const Btn_plus: string;
export const Btn_minus: string;
