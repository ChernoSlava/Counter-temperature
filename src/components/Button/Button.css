.Btn {
    width: 70px;
    height: 70px;
    border-radius: 50%;
    cursor: pointer;
    font-size: 40px;
    color: white;
    border: 1px solid white;
}

.Btn_plus {
    background-color: #696867

}
.Btn_plus:hover {
    background-color: rgb(136, 29, 29);
    transition: .7s;
}

.Btn_minus {
    background-color: #696867
}
.Btn_minus:hover {
    background-color: rgb(19, 28, 192);
    transition: .7s;
}