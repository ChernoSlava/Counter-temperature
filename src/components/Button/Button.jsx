import React from "react";
import styles from './Button.css';

const typeCss = {
    'plus': styles.Btn_plus,
    'minus': styles.Btn_minus
}

export const Button = ({ title, type }) => {
    return(
        <button className={`${styles.Btn} ${typeCss[type] || styles.Btn_plus}`}>{title}</button>
    );
}
