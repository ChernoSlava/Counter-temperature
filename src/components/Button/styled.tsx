import styled from "styled-components";

export type ButtonType = 'plus' | 'minus';

export const ButtonStyled = styled.button<{
  renderType: ButtonType
}>`
    width: 70px;
    height: 70px;
    border-radius: 50%;
    cursor: pointer;
    font-size: 40px;
    color: white;
    border: 1px solid white;

    ${props => props.renderType === 'plus' && `
        background-color: #696867;

        &:hover {
          background-color: rgb(136, 29, 29);
          transition: .7s;
        }
    `}
      ${props => props.renderType === 'minus' && `
        background-color: #696867;

        &:hover {
          background-color: rgb(19, 28, 192);
          transition: .7s;
        }
    `}
`