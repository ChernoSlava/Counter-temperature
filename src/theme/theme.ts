export const theme = {
  colors: {
    grey10: '#696867',

    blue10: 'rgb(19, 28, 192)',
    blue20: 'rgb(72, 133, 218)',
    blue30: 'rgb(98, 138, 196)',
    blue40: 'rgb(0, 69, 166)',

    red10: 'rgb(136, 29, 29)',
    red20: 'rgb(252, 38, 0)',
    red30: 'rgb(215, 104, 84)',

    black: 'black',
    white: 'white',
  },
};

export type ThemeType = typeof theme;
