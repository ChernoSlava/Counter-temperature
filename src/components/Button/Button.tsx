import React from 'react';
import { ButtonStyled, ButtonType } from './styled';

export const Button: React.FC<{
  title: string;
  type: ButtonType;
  onClick: () => void;
}> = ({ title, type, ...props }): JSX.Element => {
  return (
    <ButtonStyled renderType={type} {...props}>
      {title}
    </ButtonStyled>
  );
};
