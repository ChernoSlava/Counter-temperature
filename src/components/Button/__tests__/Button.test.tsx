/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import { render, fireEvent } from '@testing-library/react';
import React from 'react';
import { ThemeProvider } from 'styled-components';

import { theme } from '../../../theme';
import { Button, ButtonProps } from '../Button';

const getUI = (props?: ButtonProps): JSX.Element => {
  return (
    <ThemeProvider theme={theme}>
      <Button
        title="I am button"
        type="plus"
        onClick={jest.fn()}
        {...(props || {})}
      />
    </ThemeProvider>
  );
};

describe('Button tests', () => {
  it('should render correct plus button', () => {
    const { container } = render(getUI());

    expect(container).toMatchSnapshot();
  });
  it('should render correct minus button', () => {
    const { container } = render(getUI({ type: 'minus' }));

    expect(container).toMatchSnapshot();
  });

  it('should handle onClick', () => {
    const onClick = jest.fn();

    const { getByText } = render(
      getUI({
        type: 'plus',
        onClick,
      }),
    );

    const button = getByText('I am button');
    fireEvent.click(button);

    expect(onClick).toBeCalled();
    expect(onClick).toBeCalledTimes(1);
  });
});
