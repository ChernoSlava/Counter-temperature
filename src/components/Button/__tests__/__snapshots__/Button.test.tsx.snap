// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests should render correct minus button 1`] = `
.c0 {
  width: 70px;
  height: 70px;
  border-radius: 50%;
  cursor: pointer;
  font-size: 40px;
  color: white;
  border: 1px solid white;
  background-color: #696867;
}

.c0:hover {
  background-color: rgb(19,28,192);
  -webkit-transition: .7s;
  transition: .7s;
}

<div>
  <button
    class="c0"
  >
    I am button
  </button>
</div>
`;

exports[`Button tests should render correct plus button 1`] = `
.c0 {
  width: 70px;
  height: 70px;
  border-radius: 50%;
  cursor: pointer;
  font-size: 40px;
  color: white;
  border: 1px solid white;
  background-color: #696867;
}

.c0:hover {
  background-color: rgb(136,29,29);
  -webkit-transition: .7s;
  transition: .7s;
}

<div>
  <button
    class="c0"
  >
    I am button
  </button>
</div>
`;
